eks/clusterautoscaler/clusterautoscaler.go

```diff
@@ -0,0 +1,113 @@
+package clusterautoscaler
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/aws/aws-k8s-tester/eksconfig"
+	k8sclient "github.com/aws/aws-k8s-tester/pkg/k8s-client"
+	gotemplate "github.com/aws/aws-k8s-tester/pkg/util"
+)
+
+// Constants
+const (
+	NodeGroupFormatter         string = "%s-nodegroup-%d"
+	NodeGroupArgumentFormatter string = "--nodes=%d:%d:%s"
+)
+
+// ClusterAutoscaler is an addon that installs Cluster Autoscaler
+type ClusterAutoscaler struct {
+	K8sClient k8sclient.EKS
+	Config    *eksconfig.Config
+}
+
+// IsEnabled returns true if enabled
+func (c *ClusterAutoscaler) IsEnabled() bool {
+	return c.Config.Spec.ClusterAutoscaler != nil
```

Comment:
I like the idea of `nil` check here without an extra `Enable` field. How are we going to integrate this to the current env parsing function? I'd like to keep env option in case one doesn't want to write an YAML.

Reply:
On second thought, `Enable` field is kind of needed, in case an add-on does not specify any other fields (e.g. NLB hello world just needs Enable == true with other fields set to defaults).

Reply:
This can also be achieved using NLB = {}. Addon pointer existence can fill the role of "enabled"

Regarding integration with env variables, I was thinking that we could do the following order of operations.
1. Read provided YAML file into a go struct
2. Merge Addons from Env Vars into go struct
3. Call Default
4. Call Validate

WDYT?